### PR TITLE
Adds db:migrate tool, article mongoose model.

### PR DESCRIPTION
**Pull request by FieryCod**

### CODE REVIEW

**Comment by FieryCod** [comment]
Need to improve articleController tests to make getOneById and getAll independent.

File: api/swagger/parameters/mongodb.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+ID:
+  name: id
+  in: path
+  description: Parameter for proper mongodb object id
+  required: true
+  type: string
+  pattern: "^[a-zA-Z0-9]{24}$" # mongodb object id hash
+  # TODO: $ref: 'mongodb.yaml#/ObjectID'
```
Comment (k911) [inline api/swagger/parameters/mongodb.yaml]: this one doesn't work dunno why

**Comment by k911** [comment]
- [ ]  config/db.js
it's important that database credentials not be hardcoded into code, for easy installation.

File: api/swagger/parameters/mongodb.yaml (hunk 1 above)
Comment (FieryCod) [inline api/swagger/parameters/mongodb.yaml]: You mean pattern?

Comment (k911) [inline api/swagger/parameters/mongodb.yaml]: This code: `$ref: 'mongodb.yaml#/ObjectID'` doesn't work inside mongodb.yaml

Comment (FieryCod) [inline api/swagger/parameters/mongodb.yaml]: According to swagger specification it only works if parameter is defined as a "in:body" parameter.

**Pull request merged**